### PR TITLE
fix: Rename ASGIHTTPSender to BufferedASGISender for Ray compatibility.

### PR DESCRIPTION
**Pull request by HamzaFarhan**
## What does this PR address?
An update in Ray renamed ASGIHTTPSender to BufferedASGISender. This change has not been made in BentoML. I just renamed it.

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [x] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/_internal/ray/__init__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from functools import partial
 
 import requests
-from ray.serve._private.http_util import ASGIHTTPSender
+from ray.serve._private.http_util import BufferedASGISender
```
[the remark this answers is not in the document]
Comment (HamzaFarhan) [reply src/bentoml/_internal/ray/__init__.py:7]: It was merged on [June 8](https://github.com/ray-project/ray/pull/36168) which was version [2.5.0](https://github.com/ray-project/ray/releases/tag/ray-2.5.0). The current version is [2.6.3](https://github.com/ray-project/ray/releases/tag/ray-2.6.3).

[the remark this answers is not in the document]
Comment (aarnphm) [reply src/bentoml/_internal/ray/__init__.py:7]: Yes, so I think we should have backward compat for this.

[the remark this answers is not in the document]
Comment (HamzaFarhan) [reply src/bentoml/_internal/ray/__init__.py]: Added the changes.

File: src/bentoml/_internal/ray/__init__.py [hunk 2]
```diff
@@ -4,7 +4,14 @@
 from functools import partial
 
 import requests
-from ray.serve._private.http_util import BufferedASGISender
+
+from bentoml._internal.utils import pkg
+
+if pkg.pkg_version_info("ray")[1] >= 5:
```
Comment (aarnphm) [inline src/bentoml/_internal/ray/__init__.py:10]:
```suggestion
if pkg.pkg_version_info("ray")[:2] >= (2, 5):
```

File: src/bentoml/_internal/ray/__init__.py [hunk 3]
```diff
@@ -4,7 +4,14 @@
 from functools import partial
 
 import requests
-from ray.serve._private.http_util import BufferedASGISender
+
+from bentoml._internal.utils import pkg
```
Comment (aarnphm) [inline src/bentoml/_internal/ray/__init__.py:8]:
```suggestion
from ._internal.utils import pkg
```

File: src/bentoml/_internal/ray/__init__.py [hunk 4]
```diff
@@ -4,7 +4,14 @@
...
 import requests
-from ray.serve._private.http_util import ASGIHTTPSender
+
+from ._internal.utils import pkg
+
+if pkg.pkg_version_info("ray")[:2] >= (2, 5):
+    from ray.serve._private.http_util import BufferedASGISender as ASGIHTTPSender
+else:
+    from ray.serve._private.http_util import ASGIHTTPSender
+# from ray.serve._private.http_util import BufferedASGISender
```
Comment (aarnphm) [inline src/bentoml/_internal/ray/__init__.py:14]:
```suggestion
```

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
Thanks for the contribution!

**Pull request merged**